samples/snippets/src/main/java/pubsub/PublishAvroRecordsExample.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_publish_avro_records]
+import com.google.api.core.ApiFuture;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.pubsub.v1.SchemaServiceClient;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.Encoding;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.Schema;
+import com.google.pubsub.v1.SchemaName;
+import com.google.pubsub.v1.TopicName;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.Schema.*;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+
+public class PublishAvroRecordsExample {
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    // Use an existing topic with schema.
+    String topicId = "your-topic-id";
+    // Use an existing schema.
+    String schemaId = "your-schema-id";
+    // Use an Avro file that complies to the existing schema.
+    String avroFile = "path/to/an/avro/file";
+
+    publishAvroRecordsExample(projectId, topicId, schemaId, avroFile);
+  }
+
+  public static void publishAvroRecordsExample(
+      String projectId, String topicId, String schemaId, String avroFile)
+      throws IOException, ExecutionException, InterruptedException {
+
+    Encoding encoding = null;
+    Schema schema = null;
+
+    TopicName topicName = TopicName.of(projectId, topicId);
+    SchemaName schemaName = SchemaName.of(projectId, schemaId);
+
+    // Get project schema.
+    try (SchemaServiceClient schemaServiceClient = SchemaServiceClient.create()) {
+      schema = schemaServiceClient.getSchema(schemaName);
+    }
+
+    // Get topic encoding type.
+    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
+      encoding = topicAdminClient.getTopic(topicName).getSchemaSettings().getEncoding();
+    }
+
+    Publisher publisher = null;
+    try {
+      publisher = Publisher.newBuilder(topicName).build();
+
+      // Prepare to read/deserialize Avro records.
+      DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
```

Comment:
I think it's going to be much more typical to use generated code rather than the Generic objects and so I think our samples should reflect that. See https://avro.apache.org/docs/current/gettingstartedjava.html#Serializing+and+deserializing+with+code+generation.

Reply:
The workflow I had in mind is given an Avro file, publishing every Avro record from the file to Pub/Sub. But I see your point about creating a class from an Avro schema. In that case, the workflow will be:

1. user open text editor, create a JSON-formatted schema file and save as `.avsc`. 
2. download `avro-tools-1.10.1.jar`
3. run `java -jar /path/to/avro-tools-1.10.1.jar compile schema <schema file> <destination>` to create desired class
4. run our publish example, which uses the generated class